packages/core/components/FileList/useFileAccessContextMenu.ts

```diff
@@ -18,6 +19,28 @@ export default function useFileAccessContextMenu(filters?: FileFilter[], onDismi
     const fileSelection = useSelector(selection.selectors.getFileSelection);
     const userSelectedApplications = useSelector(interaction.selectors.getUserSelectedApplications);
     const { executionEnvService } = useSelector(interaction.selectors.getPlatformDependentServices);
+    const [plateLink, setPlateLink] = React.useState<string | undefined>(undefined);
+
+    fileSelection.fetchAllDetails().then((fileDetails) => {
+        // Grabbing plate barcode
+        const platebarcode = fileDetails[0].annotations.find((x) => x.name === "Plate Barcode");
+        //If theres a barcode make plateUI option available
+        if (!platebarcode?.values) {
+            const barcode = platebarcode?.values[0];
+            if (global.fileExplorerServiceBaseUrl === FileExplorerServiceBaseUrl.STAGING) {
+                setPlateLink(
+                    `http://stg-aics.corp.alleninstitute.org/labkey/aics_microscopy/AICS/editPlate.view?Barcode=${barcode}`
+                );
+            } else if (
+                global.fileExplorerServiceBaseUrl === FileExplorerServiceBaseUrl.PRODUCTION
+            ) {
+                setPlateLink(
+                    `https://aics.corp.alleninstitute.org/labkey/aics_assays/AICS/Assays/plateDetail.view?Barcode=${barcode}`
```

Comment:
It looks like if the user is on staging, they get linked to the `aics_microscopy` version of the plate UI, but if they're on prod, they get linked to the `aics_assays` one. Maybe the plate link should be generated by its own function, something like:

 ```
function createBarcodeLink(env, barcode) {
 // and the middle part of this could even be declared as a constant somewhere
 return `http://{env}.corp.alleninstitute.org/labkey/aics_microscopy/AICS/editPlate.view?Barcode=${barcode}`
}
```
Oh, and maybe double-check that the `http` vs `https` part is consistent in both

Reply:
Rather than know about the hosts at all you can pull this directly from the state [here](https://github.com/AllenInstitute/aics-fms-file-explorer-app/blob/master/packages/core/state/interaction/selectors.ts#L18).
```
const fileExplorerBaseUrl = useSelector(interactionSelectors.getFileExplorerBaseUrl);
const plateUIRef = `${fileExplorerBaseUrl}/labkey/aics_microscopy/AICS/editPlate.view?Barcode=${barcode}`
```